### PR TITLE
Add ability to store and ellipsize dynamic menu item lengths

### PR DESCRIPTION
**Pull request by tari01**
This is a base for fixing issues such as https://github.com/AyatanaIndicators/ayatana-indicator-datetime/issues/29 and https://github.com/AyatanaIndicators/ayatana-indicator-notifications/issues/18.

@sunweaver We will need Weblate set up for libayatana-common

### CODE REVIEW

File: src/utils.h [hunk 1]
```diff
@@ -47,3 +49,5 @@ gboolean ayatana_common_utils_execute_command(const gchar * cmd);
 gboolean ayatana_common_utils_open_url(const gchar * url);
 gboolean ayatana_common_utils_have_program(const gchar * program);
 gboolean ayatana_common_utils_zenity_warning(const char *icon_name, const char *title, const char *text);
+
+void ayatana_common_utils_elipsize(char *sText);
```
Comment (sunweaver) [inline src/utils.h:53]: Urgh, don't split prototype definition and code for it into two different commits. Please unite them.

File: src/utils.c [hunk 2]
```diff
@@ -273,3 +273,29 @@ ayatana_common_utils_zenity_warning (const char * icon_name,
   g_free (zenity);
   return confirmed;
 }
+
+void ayatana_common_utils_elipsize(char *sText)
```
Comment (sunweaver) [inline src/utils.c:277]: Please unite with prototype addition.

File: cmake/UseGSettings.cmake [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# GSettings.cmake, CMake macros written for Marlin, feel free to re-use them.
+
+macro(add_schema SCHEMA_NAME)
```
Comment (sunweaver) [inline cmake/UseGSettings.cmake:3]: Please don't ship yet another GSettings CMake module with this PR. Please use the GSettings module from cmake-extras. Thanks.

**Review by sunweaver — CHANGES_REQUESTED** [review]
Cool. Thanks for implementing this so meticulously and even making the string length that defines when to elipsize configurable. However, I have some feedback that requires a second iteration of development. Mainly dropping your own GSettings module and re-using what we already have in cmake-extras.

**Comment by tari01** [comment]
> Cool. Thanks for implementing this so meticulously and even making the string length that defines when to elipsize configurable. However, I have some feedback that requires a second iteration of development. Mainly dropping your own GSettings module and re-using what we already have in cmake-extras.

@sunweaver Done.

File: data/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+# org.ayatana.common.gschema.xml
+
+find_package(GSettings)
+set(ENV{LC_ALL} "C")
+execute_process(COMMAND intltool-merge -quiet --xml-style --utf8 --no-translations "${CMAKE_CURRENT_SOURCE_DIR}/org.ayatana.common.gschema.xml.in" "${CMAKE_CURRENT_BINARY_DIR}/org.ayatana.common.gschema.xml")
+add_schema("org.ayatana.common.gschema.xml")
```
Comment (sunweaver) [inline data/CMakeLists.txt:6]: This commit leaves the project in a broken state. Please commit `org.ayatana.common.gschema.xml` with it.

File: tests/CMakeLists.txt [hunk 5]
```diff
@@ -9,7 +9,7 @@ include_directories(
 add_executable(tst_utils tst_utils.cpp)
 target_link_libraries(tst_utils
     ayatana-common
-
+    ${GLIB_LIBRARIES}
```
Comment (sunweaver) [inline tests/CMakeLists.txt:12]: Why commit this and the test file where it is required separately?

Comment (sunweaver) [reply tests/CMakeLists.txt:12]: We loose context with these atomics commits.

Normally, I'd expect: code change, unit tests and build system adaptations in one commit. With some proper commit message that describes what this change does.

File: tests/tst_utils.cpp [hunk 6]
```diff
@@ -79,3 +79,55 @@ TEST_F(XdgCurrentDesktopUtilsTest, isBudgie)
     setenv("XDG_CURRENT_DESKTOP", "Budgie:GNOME", 1);
     EXPECT_TRUE(ayatana_common_utils_is_budgie());
 }
+
+class StringFunctionsTest : public ::testing::Test
```
Comment (sunweaver) [inline tests/tst_utils.cpp:83]: Please commit this together with the function the test is for.

File: CMakeLists.txt [hunk 7]
```diff
@@ -46,6 +47,8 @@ include_directories(${URLDISPATCHER_INCLUDE_DIRS})
 set(CC_WARNING_ARGS " -Wall -pedantic -Wextra -Wno-missing-field-initializers")
 
 add_subdirectory(src)
+add_subdirectory(data)
```
Comment (sunweaver) [inline CMakeLists.txt:50]: This should go with the commit that adds the CMakeLists.txt file and the .xml file.

File: CMakeLists.txt [hunk 8]
```diff
@@ -46,6 +47,8 @@ include_directories(${URLDISPATCHER_INCLUDE_DIRS})
...
+add_subdirectory(po)
```
Comment (sunweaver) [inline CMakeLists.txt:51]: This change should go with the commit that adds the po/ directory.

**Review by sunweaver — CHANGES_REQUESTED** [review]
Hmmm... well. Sorry for being so picky, but we need to improve some Git usage standards. Technically this complete PR is ok. Documentation-wise (Git commits, messages, what is committed together, etc.) this PR needs to be reworked.

Please make sure that after two years we understand what and why we did this change. The only tool we have for this is the documentation system provided by Git, i.e. the commit messages.

**Comment by tari01** [comment]
> Hmmm... well. Sorry for being so picky, but we need to improve some Git usage standards. Technically this complete PR is ok. Documentation-wise (Git commits, messages, what is committed together, etc.) this PR needs to be reworked.
> 
> Please make sure that after two years we understand what and why we did this change. The only tool we have for this is the documentation system provided by Git, i.e. the commit messages.

@sunweaver Please re-review.

**Review by sunweaver — APPROVED** [review]
LGTM now, thanks!

**Comment by sunweaver** [comment]
@tari01: I have reworked the commit messages once more and merged two of the three commits into one. But it is merged now.

**Pull request merged**